src/ng/directive/input.js

```diff
@@ -2142,7 +2142,8 @@ var ngValueDirective = function() {
    *  makes it possible to use ngValue as a sort of one-way bind.
    */
   function updateElementValue(element, attr, value) {
-    element.prop('value', value);
+    var propValue = isDefined(value) ? value : (msie === 9) ? '' : null;
```

Comment:
It is a good idea to add a comment here explaining the IE9 issue. Something like:

```js
// Support: IE9 only
// In IE9 values are converted to string (e.g. `input.value = null` results in `input.value === 'null'`).
```